app/app.go

```diff
@@ -637,18 +652,34 @@ func NewDesmosApp(
 	)
 
 	// The gov proposal types can be individually enabled
-	enabledWasmProposals := GetEnabledProposals()
-	if len(enabledWasmProposals) != 0 {
-		govRouter.AddRoute(wasm.RouterKey, wasm.NewWasmProposalHandler(app.WasmKeeper, enabledWasmProposals))
+	if len(wasmEnabledProposals) != 0 {
+		govRouter.AddRoute(wasm.RouterKey, wasm.NewWasmProposalHandler(app.WasmKeeper, wasmEnabledProposals))
 	}
 
 	app.GovKeeper = govkeeper.NewKeeper(
 		appCodec, keys[govtypes.StoreKey], app.GetSubspace(govtypes.ModuleName), app.AccountKeeper, app.BankKeeper,
 		&stakingKeeper, govRouter,
 	)
 
-	// Add wasm module route to the ibc router, then set and seal it
-	ibcRouter.AddRoute(wasm.ModuleName, wasm.NewIBCHandler(app.WasmKeeper, app.IBCKeeper.ChannelKeeper))
+	// Create IBC modules with ibcfee middleware
+	transferIBCModule := ibcfee.NewIBCMiddleware(ibctransfer.NewIBCModule(app.TransferKeeper), app.IBCFeeKeeper)
+	profilesIBCModule := ibcfee.NewIBCMiddleware(profiles.NewIBCModule(app.appCodec, app.ProfileKeeper), app.IBCFeeKeeper)
+	wasmIBCModule := ibcfee.NewIBCMiddleware(wasm.NewIBCHandler(app.WasmKeeper, app.IBCKeeper.ChannelKeeper, app.IBCFeeKeeper), app.IBCFeeKeeper)
+
+	icaControllerIBCModule := ibcfee.NewIBCMiddleware(
+		icacontroller.NewIBCMiddleware(nil, app.ICAControllerKeeper),
+		app.IBCFeeKeeper)
```

Comment:
Since there is no public ICA auth module available at the moment, so set it `nil` as it is not enabled, See:
https://ibc.cosmos.network/main/apps/interchain-accounts/integration.html#disabling-controller-chain-functionality

The demo of the controller can be found the following repo, we should implement our own or waiting for the public one released.
https://github.com/cosmos/interchain-accounts-demo